libs/vkd3d/device.c

```diff
@@ -5022,6 +5022,7 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_CreateSharedHandle(d3d12_device_if
                 if (!vkd3d_set_shared_metadata(*handle, &metadata, sizeof(metadata)))
                     ERR("Failed to set metadata for shared resource, importing created handle will fail.\n");
             }
+            CloseHandle(*handle);
```

Comment:
This looks strange. This is in CreateSharedHandle, and we're supposed to return a proper handle to the caller here, right?

Reply:
Oh yeah, sorry for the noise. Some tests would help to avoid such stupid mistakes, I think we will get there with shared resources.